src/main/java/redis/clients/jedis/JedisFactory.java

```diff
@@ -116,6 +116,20 @@ public void setHostAndPort(final HostAndPort hostAndPort) {
     this.hostAndPort.set(hostAndPort);
   }
 
+  public void setPassword(final String password) throws IllegalArgumentException {
+    if (this.user != null) {
+      throw new IllegalArgumentException();
+    }
+    this.password = password;
+  }
+
+  public void setPassword(final String user, final String password) throws IllegalArgumentException {
+    if (!java.util.Objects.equals(this.user, user)) {
```

Comment:
@sazzad16 why do we force same user?

Reply:
@gkorland In redis-cli on Redis ACL, after doing `AUTH user pass` once, if we want to `AUTH` again, we have to do `AUTH user newpass` not just `AUTH newpass`. I tried to imitate that here.
I also thought, this would help to us to avoid any mistaken password change as a wrong password (but a valid one for other user) would break the whole pool.

Reply:
Please add a JavaDoc explaining it